docs/containers-registries.conf.5.md

```diff
@@ -18,42 +18,49 @@ VERSION 2 is the latest format of the `registries.conf` and is currently in
 beta. This means in general VERSION 1 should be used in production environments
 for now.
 
-Every registry can have its own mirrors configured.  The mirrors will be tested
-in order for the availability of the remote manifest.  This happens currently
-only during an image pull.  If the manifest is not reachable due to connectivity
-issues or the unavailability of the remote manifest, then the next mirror will
-be tested instead.  If no mirror is configured or contains the manifest to be
-pulled, then the initially provided reference will be used as fallback.  It is
-possible to set the `insecure` option per mirror, too.
-
-Furthermore it is possible to specify a `prefix` for a registry.  The `prefix`
-is used to find the relevant target registry from where the image has to be
-pulled.  During the test for the availability of the image, the prefixed
-location will be rewritten to the correct remote location.  This applies to
-mirrors as well as the fallback `location`.  If no prefix is specified, it
-defaults to the specified `location`.  For example, if
-`prefix = "example.com/foo"`, `location = "example.com"` and the image will be
-pulled from `example.com/foo/image`, then the resulting pull will be effectively
-point to `example.com/image`.
-
-By default container runtimes use TLS when retrieving images from a registry.
-If the registry is not setup with TLS, then the container runtime will fail to
-pull images from the registry. If you set `insecure = true` for a registry or a
-mirror you overwrite the `insecure` flag for that specific entry.  This means
-that the container runtime will attempt use unencrypted HTTP to pull the image.
-It also allows you to pull from a registry with self-signed certificates.
-
-If you set the `unqualified-search = true` for the registry, then it is possible
-to omit the registry hostname when pulling images.  This feature does not work
-together with a specified `prefix`.
+The configuration is a sequence of `[[registry]]` entries.  Given an input image
+name, a single entry is chosen; that entry must have a `prefix` field set to be
+a prefix of the input image name (taking into account namespace/repo/tag/digest
+separators), and (only) the entry with the longest matching `prefix` field
+is used.
```

Comment:
I still find it hard to brain parse, as we immediately dive into some very specific detail. My suggestion would be to first explain the corner stones of such a table entry.

* Each entry allows us to configure specific attributes of a registry. What are these attributes?
* Location is mandatory.
* Then describe `insecure`.
* Then describe `blocked`.
* Then describe `unqualified-search`.
* Then explain `prefix` and how it can be used (and that it defaults to location).

Now we should have all pieces to explain a mirror (which also has the insecure field) and how they fit into the picture for pulling and what they mean when `unqualified-search` is set.

I also think it would be very beneficial to give examples *in line* for some of these attributes (and how they can be combined).

Reply:
> * Each entry allows us to configure specific attributes of a registry. What are these attributes?
…
> * Then describe `insecure`.
…
> I also think it would be very beneficial to give examples _in line_ for some of these attributes (and how they can be combined).

In total, that seems to suggest having everything documented 2-3 times, and 2 examples for every item (one in-line and one at the end).

This is primarily reference, authoritative, documentation; it should not repeat itself nor does it necessarily have to be useful for a complete newcomer (although it’s _good_ if it is readable for a newcomer, it’s not the _first_ priority; the _first_ priority is to be an authoritative reference).

---

Let me try coming at this from a completely different angle, see below.